Plugin/ConfigSectionPaymentPlugin.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Omise\Payment\Plugin;
+
+use Magento\Config\Model\Config as CoreConfig;
+use Omise\Payment\Model\Config\Config;
+use OmiseCapabilities;
+use OmiseAuthenticationFailureException;
+use Magento\Framework\Exception\LocalizedException;
+
+class ConfigSectionPaymentPlugin
+{
+    /**
+     * Error code sent from the API and the message to be displayed on screen
+     *
+     * @var array
+     */
+    private $errorCodes = [
+        'authentication_failure' => 'The keys for Omise Payment are invalid',
+        'key_expired_error' => 'The keys for Omise Payment are expired.'
+    ];
+
+    /**
+     * @param \Omise\Payment\Model\Config\Config $config
+     */
+    public function __construct(Config $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Magento\Config\Model\Config $coreConfig
+     */
+    public function beforeSave(CoreConfig $coreConfig)
+    {
+        if ('payment' === $coreConfig->getData('section')) {
+            $keys = $this->getKeys($coreConfig->toArray());
+
+            // if both keys are empty then we ignore the check.
+            if ($keys['public_key'] || $keys['secret_key']) {
+                try {
+                    // Fetching capabilities to check the supplied keys validity
+                    OmiseCapabilities::retrieve($keys['public_key'], $keys['secret_key']);
+                } catch (OmiseAuthenticationFailureException $e) {
+                    $errors = $e->getOmiseError();
+                    throw new LocalizedException(__($this->errorCodes[$errors['code']]));
```

Comment:
what if the error code is not part of the `errorCodes` list? should we have a default message?

Reply:
Yes, I have added a default message. I have also added other error codes that are listed under 401 and 403 of the [API errors](https://www.omise.co/api-errors).